src/constructRoutes.js

```diff
@@ -215,24 +221,47 @@ function validateAndSanitize(routesConfig) {
     routesConfig.base = "/";
   }
 
-  validateArray("routesConfig.routes", routesConfig.routes, validateRoute);
+  const pathname = inBrowser ? window.location.pathname : "/";
+  const hashPrefix = routesConfig.mode === "hash" ? pathname + "#" : "";
 
-  function validateRoute(route, propertyName) {
+  validateArray(
+    "routesConfig.routes",
+    routesConfig.routes,
+    validateRoute,
+    hashPrefix + routesConfig.base
+  );
+
+  function validateRoute(route, propertyName, parentPath) {
     validateObject(propertyName, route);
 
     if (route.type === "application") {
-      validateKeys(propertyName, route, ["type", "name"], disableWarnings);
+      validateKeys(
+        propertyName,
+        route,
+        ["type", "name", "props"],
+        disableWarnings
+      );
+      if (route.props) {
```

Comment:
This is an extra feature I added into this PR - props can now be specified on the application objects passed into constructRoutes